src/app/public/modules/auth-http-client/local-endpoint-checker.spec.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  HttpHandler
+} from '@angular/common/http';
+
+import {
+  throwError
+} from 'rxjs/index';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  LocalEndpointChecker
+} from './local-endpoint-checker';
+
+type Spy<T> = { [Method in keyof T]: jasmine.Spy; };
+
+describe('Local Endpoint Checker', () => {
+
+  let next: Spy<HttpHandler> = jasmine.createSpyObj('HttpHandler', ['handle']);
+  const prodUrl = 'https://eng-pusa01.app.blackbaud.net/hub00/version';
+  const requestUrl = '1bb://eng-hub00-pusa01:8080/version';
+
+  it('should convert to a local server url when there is a port provided and a successful local service check', (done) => {
+
+    next.handle.and.returnValue(Observable.of('ok'));
+
+    invokeUrlCheck().then((resultingUrl) => {
+      expect(resultingUrl).toBe('http://localhost:8080/version');
+      done();
+    });
+  });
+
+  it('should rely on the fallback when serving locally with a provided port but are unable to find the service locally', (done) => {
```

Comment:
technically, this test is named slightly incorrectly. the _serving locally_ check happens in the interceptor not local endpoint checker.

but more importantly, i think you need one more test for the branch when `requestUrl` does _not_ contain a port, right?

Reply:
That test is on the intercepter itself.  I can move it here but I left it in the interceptor spec to make sure we invoked all the appropriate logic as expected today.  The scenario you're describing is a backward compatibility scenario and so I left it in the existing spec.  I only add specs completely tied to the new logical paths around using a client to this checker spec.  I'm happy to move/duplicate the test but I wanted to make sure if I moved it others felt it should be moved here.

Reply:
I can update the test though.  Would you like a comment about the scenario you're calling out or should I just duplicate it?